server/tools/peertube-import-videos.ts

```diff
@@ -33,27 +34,64 @@ command
   .option('-U, --username <username>', 'Username')
   .option('-p, --password <token>', 'Password')
   .option('--target-url <targetUrl>', 'Video target URL')
+  .option('--tmpdir <tmpdir>', 'Working directory')
   .option('--since <since>', 'Publication date (inclusive) since which the videos can be imported (YYYY-MM-DD)', parseDate)
   .option('--until <until>', 'Publication date (inclusive) until which the videos can be imported (YYYY-MM-DD)', parseDate)
-  .option('-v, --verbose', 'Verbose mode')
+  .option('--first <first>', 'Process first n elements of returned playlist')
+  .option('--last <last>', 'Process last n elements of returned playlist')
+  .option('-v, --verbose [verbosity]', 'Verbosity, from 0 (only report errors) to 4 (for debugging), default is 2 (info)')
   .parse(process.argv)
 
 getServerCredentials(command)
   .then(({ url, username, password }) => {
     if (!program[ 'targetUrl' ]) {
-      console.error('--targetUrl field is required.')
+      log.error('--targetUrl field is required.')
 
       process.exit(-1)
     }
 
+    if (!program['tmpdir']) {
+      program[ 'tmpdir' ] = __dirname;
+    } else {
+      if (!existsSync(program[ 'tmpdir' ])) {
+        log.error('--tmpdir %s: directory does not exist or is not accessible', program[ 'tmpdir' ])
+
+        process.exit(-1)
+      }
+    }
+
+    log.setLevel('info')
+    if (program[ 'verbose' ]) {
+      switch (program[ 'verbose' ]) {
+        case '0':
+          log.setLevel('error')
+          break
+        case '1':
+          log.setLevel('warn')
+          break
+        case '2':
+          log.setLevel('info')
+          break
+        case '3':
+          log.setLevel('debug')
+          break
+        case '4':
+          log.setLevel('trace')
+          break
+        default:
+          log.setLevel('debug')
+          break
+      }
+    }
```

Comment:
Wouldn't be possible to use a map object? Or even an array? Like:
```js
const logLevels = ['error', 'warn', 'info', 'debug', 'trace']
const level = parseInt(program[ 'verbose' ]) > 4 ? 2 : program[ 'verbose' ]
log.setLevel(level)
```

But I wonder if we shouldn't let the user specify directly the verbosity (`error`, `info`, …) which are self-explanatory values (and raise an error if the passed value is not among the ones expected).

Reply:
This cat can be skinned in many ways, sure.